### DISCUSSION TITLE
Standalone sender - Log of the "receipt"

### DISCUSSION THREAD

**Discussion by LasseAtomic** [OxalisCommunity/oxalis#534] [Ideas]
(MOVED from the Oxalis-as4 thread)
https://github.com/difi/Oxalis-AS4/issues/131

Hi Guys

I'm new in peppol and have setup Tomcat Oxalis inbound and for now the standalone application for outbound.
But how can I log that the document is actually send, get a "proff of sending"

My goal would be a commandline respons / xml resoponsefile / easily parsed logfile / ??? that could give me

My DocumentID from SBDH
Sender, Recipient
Endpoint URL
Endpoint Certificate CN
Status Success or Failure
Can I get that somehow from the standalone sender?

AS I see it.

In the command line it says:
"6cccc3d3-6451-41c4-9fdc-487453e3e1e4@EDI01 transmission took 4273 ms
Attempted to send 1 files
Failed transmissions: 0"
So here I can see the GUID and that is was ok (1 attempt, 0 errors), but hard to parse. And 100 attempted, 1 failed is a bad message to get :-)

I get a .receipt.dat file.
This file includes MY signature, sender and reciever id, but nothing about the endpoint url, endpoint signature or my document id. So what does this mean?? A file was sent ok?? Do I even know it is sent ok with this file? The root xml name confuses me "DeliveryNonDeliveryToRecipient"

In my log file I get info about endpoint and payload
Address: AS4-URL
Content-Type: application/soap+xml;charset=UTF-8
ResponseCode: 200
ExchangeId: d48e79af-608c-4625-9f08-8160696ddd9d
ServiceName: outbound-service
PortName: port
PortTypeName: DummyImpl
Headers:
Payload:
So here I can see the endpoint, but stil not my message id..not even in the message payload

**Comment by aaron-kumar** [comment]
If you want to see/store details like ap, timestamp, direction, sender, receiver, docType, profile & channel (AS2/AS4), implement/integrate statistics module (_part of Oxalis Legacy module, and may not be maintained further_). 
You can further enhance/extend it according to your requirement. Also feel free to contribute if you want to share additional implementation with community. 

P.S. moving this topic to discussion...
